pkg/config/check.go

```diff
@@ -0,0 +1,463 @@
+			}
+
+			return opts, nil
+		},
+	},
+	"settlements": {
+		NewAction: settlements.NewCheck,
+		NewOptions: func(checkGlobalConfig CheckGlobalConfig, check Check) (interface{}, error) {
+			checkOpts := new(struct {
+				DryRun             *bool          `yaml:"dry-run"`
+				ExpectSettlements  *bool          `yaml:"expect-settlements"`
+				FileName           *string        `yaml:"file-name"`
+				FileSize           *int64         `yaml:"file-size"`
+				PostageAmount      *int64         `yaml:"postage-amount"`
+				PostageDepth       *uint64        `yaml:"postage-depth"`
+				PostageWait        *time.Duration `yaml:"postage-wait"`
+				Seed               *int64         `yaml:"seed"`
+				Threshold          *int64         `yaml:"threshold"`
+				UploadNodeCount    *int           `yaml:"upload-node-count"`
+				WaitBeforeDownload *int           `yaml:"wait-before-download"`
+			})
+			if err := check.Options.Decode(checkOpts); err != nil {
+				return nil, fmt.Errorf("decoding check %s options: %w", check.Type, err)
+			}
+			opts := settlements.NewDefaultOptions()
+
+			if err := applyCheckConfig(checkGlobalConfig, checkOpts, &opts); err != nil {
+				return nil, fmt.Errorf("applying options: %w", err)
+			}
+
+			return opts, nil
+		},
+	},
+	"smoke": {
+		NewAction: smoke.NewCheck,
+		NewOptions: func(checkGlobalConfig CheckGlobalConfig, check Check) (interface{}, error) {
+			checkOpts := new(struct {
+				Bytes           *int           `yaml:"bytes"`
+				NodeGroup       *string        `yaml:"node-group"`
+				Runs            *int           `yaml:"runs"`
+				Seed            *int64         `yaml:"seed"`
+				Timeout         *time.Duration `yaml:"timeout"`
+				UploadNodeCount *int           `yaml:"upload-node-count"`
+			})
+			if err := check.Options.Decode(checkOpts); err != nil {
+				return nil, fmt.Errorf("decoding check %s options: %w", check.Type, err)
+			}
+			opts := smoke.NewDefaultOptions()
+
+			if err := applyCheckConfig(checkGlobalConfig, checkOpts, &opts); err != nil {
+				return nil, fmt.Errorf("applying options: %w", err)
+			}
+
+			return opts, nil
+		},
+	},
+	"soc": {
+		NewAction: soc.NewCheck,
+		NewOptions: func(checkGlobalConfig CheckGlobalConfig, check Check) (interface{}, error) {
+			checkOpts := new(struct {
+				PostageAmount  *int64         `yaml:"postage-amount"`
+				PostageDepth   *uint64        `yaml:"postage-depth"`
+				PostageWait    *time.Duration `yaml:"postage-wait"`
+				RequestTimeout *time.Duration `yaml:"request-timeout"`
+			})
+			if err := check.Options.Decode(checkOpts); err != nil {
+				return nil, fmt.Errorf("decoding check %s options: %w", check.Type, err)
+			}
+			opts := soc.NewDefaultOptions()
+
+			if err := applyCheckConfig(checkGlobalConfig, checkOpts, &opts); err != nil {
+				return nil, fmt.Errorf("applying options: %w", err)
+			}
+
+			return opts, nil
+		},
+	},
+}
+
+// applyCheckConfig merges given, global and default check options
```

Comment:
maybe `applyCheckConfig merges global and local options into given opts

Reply:
updated